src/screens/ChatScreen/ChatScreen.tsx

```diff
@@ -23,29 +24,47 @@ const renderBubble = ({
   child: ReactNode;
   message: MessageType.Any;
   nextMessageInGroup: boolean;
-}) => (
-  <Bubble
-    child={child}
-    message={message}
-    nextMessageInGroup={nextMessageInGroup}
-  />
-);
+}) => {
+  // Check for our special loading message
+  if (message.id === 'loading-indicator') {
```

Comment:
At first glance, the current approach of introducing a placeholder message feels like a workaround. I wonder if we can find a more direct solution, such as introducing a `showLoading`, `isThinking` or `isTyping` state that we pass to `ChatView`. I’m unsure how this would be implemented, but I’d love to explore this direction and see if it leads to a cleaner approach. wdyt?

Reply:
That makes sense this was a hackish approach. We already have inferencing and streaming so used them to create a prop called isThinking, based on this changed the Loading bubble view state.